Design/reflectometry-idf.md

```diff
@@ -0,0 +1,62 @@
+# Reflectometry IDF
+
+## Motivation
+
+The reflectometry instrument scientists do not currently use an IDF and are heavily reliant on their own scripts.
+If the IDF could properly describe their setup then it would solve existing issues and improve their workflow.
+
+Potential benefits include, but are not limited to:
+  - Correct geometric description of detector components and calculation of angles to be fed into [algorithms](https://github.com/mantidproject/mantid/issues/26971)
+  - Complex beam paths to allow for accurate description of mirror action, with a view to accurately describe the effect of gravity
```

Comment:
Possibly more accurate, though beam paths will still probably just provide an approximation (point to point per component).

Reply:
Did @gvardany do some work on a similar issue recently? I think it would be best to leave it as a point to point approximation and handle gravity corrections elsewhere but I don't know how that would be implemented. From what I understand, there are sometimes multiple bounces off mirrors before reaching the detector so these will all affect the final angles that are passed into the algorithms.

Reply:
There was an attempt by my colleague to implement a general gravity correction algorithm https://github.com/mantidproject/mantid/pull/26023 
However this is stale and outdated, so trying to recover it would be quite painful.
Besides it has never been properly validated. I'm tempted to presume, that redoing from scratch might actually be faster. Yet it is not at all trivial in case of multiple mirrors...